### PR TITLE
Adds in basic styling for post comment form

### LINKED ISSUE #26865: Post Comments Form Block: Block should ship with default styles
The post comments form block currently renders completely unstyled: 

Screen Shot 2020-11-10 at 1 40 57 PM

This should ship with a set of default styles so it can be used out of the box in block-based themes.

### PR DESCRIPTION
**Pull request by karmatosed**
This is a reworked version of stale PR #27009 which I closed today.

@kjellr and @jffng looping you in as know tracking this for Twenty Twenty Two.

@colorful-tones I removed all the review points you noted to reduce back as much as I can on styling.

Whilst I do think more styling could be done by default, I have concerns setting any font sizes or anything else at this point, so I really went as minimal as possible. I am open to other options being brought in though as it looks 'simple' to say the least.

2021-10-16 at 12 34

### CODE REVIEW

**Review by jffng — COMMENTED** [review]
What do you think about adding padding to the text inputs, and allowing the textarea to inherit the font family from the body?

Using Twenty Twenty-Two as an example:

Before | After
------- | -------
Screen Shot 2021-10-18 at 2 23 30 PM | Screen Shot 2021-10-18 at 2 24 34 PM 

I think the cookie consent text could also be smaller by default, but that's less important to me than the padding and font family on the comment textarea.

**Comment by carolinan** [comment]
It would be good if the padding matched other input fields. The search block fields has `padding: 8px;`

**Comment by kjellr** [comment]
@karmatosed I hope you don't mind, I pushed a few small updates. 

- I set the font size to `1em` to inherit the font size of the surrounding text.
- I added padding (I used the default value from button blocks — I opted not to use the `8px` from the search block because it seemed small, and that block [already doesn't provide decent default styles](https://github.com/WordPress/twentytwentytwo/issues/11)).
- I removed an errant border that was being imposed on the submit button.
- Some of these styles needed to exist both in the Post Comments Form block, as well as the Post Comments block, since those can be used independently. 
- I added just a tiny bit of space underneath the labels. This is probably the most opinionated thing I did, but it's really subtle. 

Here's how those look now: 

Logged Out|Logged In
---|---
Screen Shot 2021-10-20 at 1 12 54 PM|Screen Shot 2021-10-20 at 1 13 04 PM

**Comment by kjellr** [comment]
Here's how the updates look in TT1 Blocks too. There's some slightly unfortunate text wrapping next to the checkbox down below. Maybe we should style that too?

Logged Out|Logged In
---|---
Screen Shot 2021-10-20 at 1 54 36 PM|Screen Shot 2021-10-20 at 1 54 11 PM

**Comment by kjellr** [comment]
I tidied up that cookies consent area. In general it's looking ok. I'm unclear if this is useful in the context of https://github.com/WordPress/gutenberg/pull/35231 though?

Anyway, here's how this PR currently looks in various themes: 

Twenty Twenty-Two:

Logged Out|Logged In
---|---
Screen Shot 2021-10-21 at 8 26 45 AM|Screen Shot 2021-10-21 at 8 26 31 AM

TT1 Blocks: 

Logged Out|Logged In
---|---
Screen Shot 2021-10-21 at 8 26 06 AM|Screen Shot 2021-10-21 at 8 26 15 AM

Twenty Twenty-One:

Logged Out|Logged In
---|---
Screen Shot 2021-10-21 at 8 24 41 AM|Screen Shot 2021-10-21 at 8 24 48 AM

Twenty Twenty:

Logged Out|Logged In
---|---
Screen Shot 2021-10-21 at 8 25 22 AM|Screen Shot 2021-10-21 at 8 25 15 AM

**Comment by jffng** [comment]
A few thoughts: 

- Since prior default themes do not use blocks in their templates to render comments + comments form, I think it's safer to add these styles.
- It sounds like `post-comments` may be deprecated once the `comments-query` (#35231) and its sub-blocks are complete. But as far as I can tell, that block is not milestone'd for 11.9 / 5.9, so the styles are useful to 2022 and any other theme that uses the `post-comments` block until then.
- The post comments form is out-of-scope for the Comments query loop work, so the styles you're adding will be useful until that work is picked up.

File: packages/block-library/src/post-comments-form/style.scss [hunk 1]
```diff
@@ -1,9 +1,48 @@
...
+	input[type="submit"] {
+		border: none;
+		box-shadow: none;
+		cursor: pointer;
+		display: inline-block;
+		text-align: center;
+		overflow-wrap: break-word;
+	}
+
+	textarea,
```
Comment (jffng) [inline packages/block-library/src/post-comments-form/style.scss:13]: What if we let the `font-family: inherit;` for this textarea?

Screen Shot 2021-10-21 at 10 09 46 AM

**Review by jffng — COMMENTED** [review]
One comment about the font family but otherwise this LGTM.

File: packages/block-library/src/post-comments-form/style.scss (hunk 1 above)
Comment (kjellr) [reply packages/block-library/src/post-comments-form/style.scss:13]: Whoops — I missed that originally. Should be all set as of 4881e18.

**Comment by carolinan** [comment]
> I'm unclear if this is useful in the context of #35231 though?

I think it is still useful, I don't think the form is going to be split yet.

**Comment by carolinan** [comment]
_That spacing in Twenty Twenty though 🤐_

**Review by carolinan — APPROVED** [review]

**Comment by kjellr** [comment]
Thanks, everyone!

**Comment by karmatosed** [comment]
> I hope you don't mind, I pushed a few small updates.

@kjellr I absolutely don't mind at all and love this thank you.

**Pull request merged**